cat/opencl-herd.cat

```diff
@@ -0,0 +1,153 @@
+OpenCL
+(* OpenCL Memory Model *)
+
+(*
+* This model is based on:
+* https://github.com/herd/herdtools7/blob/master/herd-www/cat_includes/opencl_base.cat
+* https://github.com/herd/herdtools7/blob/master/herd-www/cat_includes/opencl.cat
+*)
+
+// Base relations:
+// int: same work-item
+// swg: same work-group
+// sdv: same device
+// syncbar: same barrier id
+
+// Tags:
+// WI: work-item scope
+// WG: work-group scope
+// DV: device scope
+// ALL: all-svm-devices scope
+// GLOBAL: global memory
+// LOCAL: local memory
+// EF: entry fence of barrier
+// XF: exit fence of barrier
+
+include "basic.cat"
+
+let symm(r) = r | r^-1
+let wi = int
+
+(* Inclusive scopes *)
+let incl = (wi & (WI * WI)) |
+    (swg & (WG * WG)) |
+    (sdv & (DV * DV)) |
+    (ALL * ALL)
+
+(*******************)
+(* Synchronisation *)
+(*******************)
+
+let Acq = (ACQ | SC | ACQ_REL) & (R | F | RMW)
+let Rel = (REL | SC | ACQ_REL) & (W | F | RMW)
+
+(* Release sequence *)
+let rs = (_ * RMW) | wi
+let max_co = co? & rs & ~((co & ~rs) ; co)
```

Comment:
Just a side-note: I'm not a big fan of the complementation if differences are enough: `a & ~b = a \ b`.
I know that herd's version uses the complementation so we can keep it here, but maybe we should consider adding a Wmm transformation pass to rewrite the model automatically.
Of course, the pass can be added in a different PR. I just wanna mention it here :).